File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.js
function solution(score) {
    let avg = score.map(arr => (arr[0]+arr[1])/2);
    let sorted = [...avg].sort((a,b)=>b-a);
    console.log(avg);
    
    let rank = 1;
    let temp = {};
    for(let i=0; i<sorted.length; i++){
        if(sorted[i] == sorted[i-1]){
            temp[sorted[i]] = temp[sorted[i-1]];
        }else{
            temp[sorted[i]] = rank;
        }
        rank++;
    }
    console.log(temp);
    
    let answer = avg.map(arr => temp[arr]);
    return answer;
}

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.py
def solution(score):
    arr = []
    for idx,s in enumerate(score):
        avg = (s[0]+s[1])/2
        arr.append((avg,idx))
    arr.sort(reverse = True)
    print(arr)
    
    ranks = [0]*len(score)
    prev_score = None
    prev_rank = 0
    for rank,(avg,idx) in enumerate(arr):
        if avg != prev_score:
            prev_score = avg
            prev_rank = rank+1
        ranks[idx] = prev_rank
    return ranks